pkgs/development/python-modules/gguf/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, numpy
+, poetry-core
+, pythonOlder
+}:
+buildPythonPackage rec {
+  pname = "gguf";
+  version = "0.6.0";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-suIuq6KhBsGtFIGGoUrZ8pxCk1Fob+nXzhbfOaBgfmU=";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+    poetry-core
+  ];
+
+  doCheck = false;
+
+  meta = with lib; {
```

Comment:
Missing `meta.mainProgram`.

Reply:
I think the intent for the module is to be primarily used as a dependency within other python tools - afaik, while there are three small utility scripts included, none of them are the "main program" so to speak. Here's the pypi page, with short readme: https://pypi.org/project/gguf/. Let me know if you'd still like me to specify one of them.